woocommerce/woocommerce-ios
# Shipping Labels: Update default item details for customs form

Part of #4687

# Description
This PR update default item details form custom form based on fetched products and variations. Item details should now display correct data.

# Changes
- Updated ShippingLabelsCustomsForm quantity property type to Decimal to match with order item's for safety.
- Customs form list view model: 
   - Fetch currency from order details, then send the currency all the way down to item detail form view models to display on the UI.
   - Reuse `ShippingLabelPackageDetailsResultsControllers` to fetch products and variations.
   - Observe changes of products and variations to update default item details. 
- Updated item details view model to fetch weight unit from `shippingSettingsService` to display on the UI.

# Demo
![item-details](https://user-images.githubusercontent.com/5533851/129357344-741f935c-5202-4435-a521-3aa617c1be06.gif)

# Testing steps
1. Make sure that your test store has WCShip plugin installed and package details set up.
2. Navigate to Orders tab and select an order with international shipping (origin country different from destination).
3. Select Create Shipping Label.
4. Skip through Ship To, Ship From, Package Details to configure Customs.
5. Notice that in Package Details section, item details are displayed correctly: description, weight, value, weight unit, currency. Origin country is set the same as your site address country by default.

Update release notes:
